### ISSUE TITLE
Remove RecentFilesService from Engine

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit-console#874]
In testcentric/testcentric-gui#582 we agreed that both projects would remove RecentFiles from the engine as it has nothing special to do with running tests. This is a breaking change.

**Comment by CharliePoole** [comment]
@ChrisMaddock @jnm2 @rprouse @mikkelbu @OsirisTerje 

This is ready to go to a PR although I'm holding off for now until #896 is merged.

That means we're at a decision point, so I've tried to mention everyone who may be impacted.

If we treat this as a breaking change requiring a 4.0 release, merging it to master basically implies that there will be no more 3.x releases of the engine. That has been the plan, of course, but now it's time to be sure we are ready to commit to that.

Options... 

1. Merge to master and make the next release be 4.0

2. Merge to master, but make the next release 3.13, treating this as non-breaking or at least not badly breaking.

3. Merge but postpone the decision, which implies that (2) is acceptable as well as (1).

If we are going the route of (2), then I'll advance the version in the build script to 4.0 in addition to what I've done so far.

**Comment by rprouse** [comment]
In the framework, I made a `v3.13` branch in case we need to do bug fix releases before I made any breaking changes. I could have branched off the release tag in the future, but I thought it was more explicit and will keep a 3.13 dev branch available if we need to do multiple releases and more work.

Should we do the same here and prepare for 4.0?

**Comment by ChrisMaddock** [comment]
This should be in 4.0 in my opinion.

My plan was to create a 4.0 branch and leave 3.13 as the master, merging 4.0 changes into the 4.0 branch, and eventually into master. So essentially the same as Rob's, but with a different default. 

We already have a bunch of non-breaking changes merged into master for 3.13 - I think it might be wise to keep the breaking ones separate for the time being, as we're not really clear on how long 4.0 will take. How do we feel about a separate 4.0 branch, and merging master into it now and again? Probably makes sense to merge in the 'big' PRs like the copyright header change before we create this branch.

Charlie - if everyone agrees, would you mind setting this up with your PR? As it looks this this will be the first 4.0-specific PR!

**Comment by CharliePoole** [comment]
Marked as blocked by #896

**Comment by CharliePoole** [comment]
This was merged to dev-4.0 and closed. I'm reopening it so as to replicate it in main.
